pkg/terraform/provider_runner.go

```diff
@@ -33,9 +35,13 @@ const (
 	errFmtTimeout = "timed out after %v while waiting for the reattach configuration string"
 
 	// an example value would be: '{"registry.terraform.io/hashicorp/aws": {"Protocol": "grpc", "ProtocolVersion":5, "Pid":... "Addr":{"Network": "unix","String": "..."}}}'
-	envReattachConfig = "TF_REATTACH_PROVIDERS"
-	regexReattachLine = envReattachConfig + `='(.*)'`
-	reattachTimeout   = 1 * time.Minute
+	fmtReattachEnv     = `{"%s":{"Protocol":"grpc","ProtocolVersion":%d,"Pid":%d,"Test": true,"Addr":{"Network": "unix","String": "%s"}}}`
+	fmtSetEnv          = "%s=%s"
+	envReattachConfig  = "TF_REATTACH_PROVIDERS"
+	envMagicCookie     = "TF_PLUGIN_MAGIC_COOKIE"
+	defaultMagicCookie = "d602bf8f470bc67ca7faa0386276bbdd4330efaf76d1a219cb4d6991ca9872b2"
```

Comment:
Could you add a comment about how we found this value?

Reply:
Done.